src/components/input.js

```diff
@@ -230,15 +230,17 @@ const AutocompleteSuggestions = ({ target: targetId, containerProps, children })
 }
 
 const withAutocomplete = WrappedComponent => ({
-  instructions, value, onChange, onPick, suggestions: rawSuggestions, style, id, labelId,
-  renderSuggestion = _.identity, openOnFocus = true, placeholderText, ...props
+  instructions, itemToString, value, onChange, onPick, suggestions: rawSuggestions, style, id, labelId,
+  renderSuggestion = _.identity, openOnFocus = true, suggestionFilter = Utils.textMatch, placeholderText, ...props
 }) => {
   Utils.useLabelAssert('withAutocomplete', { id, 'aria-labelledby': labelId, ...props, allowId: true })
 
-  const suggestions = _.filter(Utils.textMatch(value), rawSuggestions)
+  const suggestions = _.filter(suggestionFilter(value), rawSuggestions)
+  const controlProps = itemToString ? { itemToString: v => v ? itemToString(v) : value } : { selectedItem: value }
```

Comment:
`itemToString` is a downshift property, you cannot have both `selectedItem` and `itemToString` otherwise you will get [react control](https://reactjs.org/docs/uncontrolled-components.html) warnings

Reply:
My eyes found it a bit difficult to parse the nested ternary on the same line. Something like the below would help:

```suggestion
  const controlProps = itemToString ?
    { itemToString: v => v ? itemToString(v) : value } :
    { selectedItem: value }
```